sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -254,3 +255,184 @@ def _predict_interval(self, fh, X, coverage):
         final_interval_predictions = pandas.concat(interval_predictions, copy=False)
 
         return final_interval_predictions
+
+
+class StatsForecastBackAdapter(BaseForecaster):
```

Comment:
I think the backadapter should inherit from the `statsforecast` forecaster rather than the `sktime` one? If it makes a difference.

Reply:
Ok, I changed it to inherit from `statsforecast` base forecaster. In both nosoftdeps and nodevdeps, however, it raises exceptions of `statsforecast` not found. Do I make the same check that I did in this commit? -> [update get_test_params StatsForecastMSTL](https://github.com/sktime/sktime/pull/4865/commits/d41565e48ca076b43b724a7ddf238bd9a01e75ef)
Further, did that commit match what you meant by using `_check_soft_dependencies` for nosoftdeps exception?

Reply:
> In both nosoftdeps and nodevdeps, however, it raises exceptions of statsforecast not found.

Ah, that has to do with checks for dependency isolation when `statsforecast` is not present.

Imagine a user of `sktime` who has no dependencies (and no `statsforecast`, they would now experience an import exception upon using `sktime`, because this one module imports `statsforecast`. We have to isolate soft dependencies, to avoid users of `sktime` without all the dependencies installed get a crash.

The general procedure is to ensure `statsforecast` gets imported only if it is in the environment. Estimators take care of this in the base class automatically if we use the tags correctly.

This is now an unusual case though, since we import a class and inherit from it, so we need to get a little creative.

Here's one way we could do it, although it might not be the bets way (it is not a standard situation).

1. introduce a preamble using `_check_soft_dependencies` (with severity being `"none"`). If `statsforecast` is present, import the class. If not, define a class of the same name.
2. this will result in the class we want if `statsforecast` is present, and a class inheriting from a dummy that does not import `statsforecast` on module load. Now this class will likely be broken, but it will also not be used anywhere (because `statsforecast` is missing).

What do you think about this solution? It is similar to what we are doing for `numba`/.
`numba` uses decorators heavily, so they sit flat in the module. If `numba` is not present, we replace the decorators by dummies.

Reply:
Gotcha, that solution makes a lot of sense. Thanks!